library/ssl_msg.c

```diff
@@ -5593,8 +5593,10 @@ int mbedtls_ssl_read(mbedtls_ssl_context *ssl, unsigned char *buf, size_t len)
     n = (len < ssl->in_msglen)
         ? len : ssl->in_msglen;
 
-    memcpy(buf, ssl->in_offt, n);
-    ssl->in_msglen -= n;
+    if (len != 0) {
+        memcpy(buf, ssl->in_offt, n);
+        ssl->in_msglen -= n;
+    }
```

Comment:
Not a blocker. 

What is the defined behaviour for `mbedtls_platform_zeroize(void *buf, 0 )` ? If that is not guaranteed to be a no-op for zero len, it may be better to handdle it in the if, else block

~~~~~
    if (len != 0) {
        memcpy(buf, ssl->in_offt, n);
        ssl->in_msglen -= n;
       mbedtls_platform_zeroize(ssl->in_offt, n);
    }
~~~~~~